snuba/datasets/storages/functions.py

```diff
@@ -0,0 +1,93 @@
+from typing import List
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Array,
+    Column,
+    ColumnSet,
+    DateTime,
+    Float,
+)
+from snuba.clickhouse.columns import SchemaModifiers as Modifiers
+from snuba.clickhouse.columns import String, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.functions_processor import FunctionsMessageProcessor
+from snuba.datasets.schemas.tables import TableSchema, WritableTableSchema
+from snuba.datasets.storage import ReadableTableStorage, WritableTableStorage
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.table_storage import build_kafka_stream_loader_from_settings
+from snuba.query.processors.table_rate_limit import TableRateLimit
+from snuba.utils.schemas import SchemaModifiers
+from snuba.utils.streams.topics import Topic
+
+common_columns: List[Column[SchemaModifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("transaction_name", String()),
+    Column("timestamp", DateTime()),
+    Column("depth", UInt(32)),
+    Column("parent_fingerprint", UInt(64)),
+    Column("fingerprint", UInt(64)),
+    Column("name", String()),
+    Column("package", String()),
+    Column("path", String()),
+    Column("is_application", UInt(8)),
+    Column("platform", String()),
+    Column("environment", String(Modifiers(nullable=True))),
+    Column("release", String(Modifiers(nullable=True))),
+    Column("os_name", String()),
+    Column("os_version", String()),
+    Column("retention_days", UInt(16)),
+]
+
+raw_storage = WritableTableStorage(
+    storage_key=StorageKey.FUNCTIONS_RAW,
+    storage_set_key=StorageSetKey.FUNCTIONS,
+    schema=WritableTableSchema(
+        storage_set_key=StorageSetKey.FUNCTIONS,
+        local_table_name="functions_raw_local",
+        dist_table_name="functions_raw_dist",
+        columns=ColumnSet(
+            common_columns
+            + [
+                Column("durations", Array(Float(64))),
+                Column("profile_id", UUID()),
+                Column("materialization_version", UInt(8)),
+            ]
+        ),
+    ),
+    query_processors=[],
+    stream_loader=build_kafka_stream_loader_from_settings(
+        processor=FunctionsMessageProcessor(),
+        default_topic=Topic.PROFILES_FUNCTIONS,
+    ),
```

Comment:
Do you want to add a DLQ? (It would catch exceptions raised by the message processor and you can choose what to do with invalid messages so the consumer doesn't crash loop)

Here's an example stream loader configured with a DLQ:
https://github.com/getsentry/snuba/blob/c033f975aef88f69eaa1a405acf8285507b1e413/snuba/datasets/storages/generic_metrics.py#L166-L170

If you'd like invalid messages to be forwarded to a new dead-letter topic for this consumer, please leave a message in #discuss-eng-sns and we can set up/configure.